shared/src/netlink.rs

```diff
@@ -43,14 +46,31 @@ fn netlink_call(
         ));
     }
 
-    let n_received = socket.recv(&mut buf[..], 0).unwrap();
-    let response = NetlinkMessage::<RtnlMessage>::deserialize(&buf[..n_received])
-        .map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e))?;
-    log::trace!("netlink response: {:?}", response);
-    if let NetlinkPayload::Error(e) = response.payload {
-        return Err(e.to_io());
+    let mut responses = vec![];
+    loop {
+        let n_received = socket.recv(&mut buf[..], 0).unwrap();
+        let mut offset = 0;
+        loop {
+            let bytes = &buf[offset..];
+            let response = NetlinkMessage::<RtnlMessage>::deserialize(bytes)
+                .map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e))?;
+            responses.push(response.clone());
+            log::trace!("netlink response: {:?}", response);
+            match response.payload {
+                // We've parsed all parts of the response and can leave the loop.
+                NetlinkPayload::Ack(_) | NetlinkPayload::Done => return Ok(responses),
```

Comment:
Do we want the final `Ack`/`Done` be part of the returned `responses`? (order of the `.push()` vs. the `match {}`)

Reply:
That's probably most correct, I'll do that.

Reply:
It should be in the right order already unless I'm misreading